repos/system_upgrade/common/actors/checksaphana/libraries/checksaphana.py

```diff
@@ -195,32 +199,39 @@ def version2_check(info):
 
 
 def platform_check():
-    """ Creates an inhibitor report in case the system is not running on x86_64 """
-    if not architecture.matches_architecture(architecture.ARCH_X86_64):
-        if version.get_target_major_version() == '8':
-            elink = reporting.ExternalLink(
-                url='https://access.redhat.com/solutions/5533441',
-                title='How do I upgrade from Red Hat Enterprise Linux 7 to Red Hat Enterprise Linux 8 with SAP HANA')
-        else:
-            elink = reporting.ExternalLink(
-                url='https://access.redhat.com/solutions/6980855',
-                title='How to in-place upgrade SAP environments from RHEL 8 to RHEL 9')
-
-        reporting.create_report([
-            reporting.Title('SAP HANA upgrades are only supported on X86_64 systems'),
-            reporting.Summary(
-                ('Upgrades for SAP HANA are only supported on X86_64 systems.'
-                 ' For more information please consult the documentation.')
-            ),
-            reporting.Severity(reporting.Severity.HIGH),
-            reporting.Groups([reporting.Groups.SANITY]),
-            reporting.Groups([reporting.Groups.INHIBITOR]),
-            reporting.Audience('sysadmin'),
-            elink,
-        ])
-        return False
+    """ Creates an inhibitor report in case the SAP HANA system is not running
+    - on x86_64 when upgrding from RHEL7 to RHEL8
+    - on x86_64 or ppc64le when upgrading from RHEL8 to RHEL9 """
+    target_major_version = version.get_target_major_version()
+    arch = api.current_actor().configuration.architecture
+    if target_major_version == '8' and arch not in SAP_HANA_7TO8_ARCH:
+        elink = reporting.ExternalLink(
+            url='https://access.redhat.com/solutions/5154031',
+            title='How to in-place upgrade SAP environments from RHEL 7 to RHEL 8')
+        supported_arch = ','.join(SAP_HANA_7TO8_ARCH)
+    elif target_major_version == '9' and arch not in SAP_HANA_8TO9_ARCH:
+        elink = reporting.ExternalLink(
+            url='https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_sap_solutions/9/html/how_to_in-place_upgrade_sap_environments_from_rhel_8_to_rhel_9/index',  # noqa: E501; pylint: disable=line-too-long
```

Comment:
Please remove #noqa: E501 , it is possible to refactor that without silencing the linter.

Reply:
@fernflower  @alexxa  tbh, in case of URLs I am ok with the use of noqa, as wrapped URLs are not so fast to open from terminal (I need to copy&paste instead of just click on the link). so i have not strong opinion here. also, it's possible to use the  RH URL shortener to create public URL (I can create the shorted URL if wanted)

Reply:
Let's have the shortened url then, thanks!

Reply:
FYI, there are more long URLs in this repo, and #noqa: E501 silencer next to them. Maybe consider to refactor them too if their presence is not desired.

Reply:
Oh, nice catch. Please also fix the other urls as well.